src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -178,9 +188,19 @@ private static async Task CheckExceptionForBadConnectionStringHelperAsync(string
             catch (DataApiBuilderException ex)
             {
                 // use contains to correctly cover db/user unique error messaging
-                Assert.IsTrue(ex.Message.Contains(DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE));
-                Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
+                // if sw is not null it holds the error messaging
+                string error = sw is null ? ex.Message : sw.ToString();
+                Assert.IsTrue(error.Contains(DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE));
                 Assert.AreEqual(DataApiBuilderException.SubStatusCodes.ErrorInInitialization, ex.SubStatusCode);
+                if (sw is not null)
+                {
+                    Assert.AreEqual(HttpStatusCode.InternalServerError, ex.StatusCode);
```

Comment:
Why is this an internal server error? Why not serviceunavailable?
Did the server not come up due to invalid user input or something wrong with the server?

Reply:
This is the exception that comes back trying to invoke `GetConfig()` through the `RuntimeConfigProvider` in the case where trying to load the known config results in that config not loading but instead being `null`. In this case we had decided on an `InternalServerError`. I think this makes sense because it is indicating back to the user that the problem is with our internal processing, in this case of the config file.

![image](https://github.com/Azure/data-api-builder/assets/93220300/9bfd2b47-d366-4fd1-9c89-b3130e3e567b)

Reply:
after discussion we decided to use `ServiceUnavailable` for all of the config related errors. this is the status code used in the config validator already, so changing this matches the rest of the messaging we use.

Reply:
Since we decided to go with ServiceUnavailable, shouldn't the assert here be changed?

Reply:
yes, I had this locally only, pushed the changes now.